File: Java-Kotlin/src/main/kotlin/hashmap/N-Repeated Element in Size 2N Array.kt
package hashmap


/**
 * https://leetcode.com/problems/n-repeated-element-in-size-2n-array/
 * */

fun repeatedNTimes(nums: IntArray): Int {
    val n = nums.size / 2
    val map = HashMap<Int, Int>()
    nums.forEach {
        map[it] = map.getOrDefault(it, 0) + 1
        if (map[it] == n) return it
    }
    return 0
}

File: C++/Hash Map/N-Repeated Element in Size 2N Array.cpp
#include <bits/stdc++.h>

using namespace std;

/**
 * https://leetcode.com/problems/n-repeated-element-in-size-2n-array/
 * */

class Solution {
public:
    int repeatedNTimes(vector<int> &nums) {
        unordered_set<int> seen;
        for (int &num:nums) {
            if (seen.count(num)) return num;
            seen.insert(num);
        }
        return -1;
    }
};